iceoryx_hoofs/source/posix_wrapper/access_control.cpp

```diff
@@ -104,50 +103,38 @@ AccessController::createACL(const int32_t numEntries) noexcept
     return cxx::success<smartAclPointer_t>(reinterpret_cast<acl_t>(aclInitCall->value), freeACL);
 }
 
-bool AccessController::addPermissionEntry(const Category category,
-                                          const Permission permission,
-                                          const permissionString_t& name) noexcept
+bool AccessController::addUserPermission(const Permission permission, const PosixUser::userName_t& name) noexcept
 {
-    switch (category)
-    {
-    case Category::SPECIFIC_USER:
+    if (name.empty())
     {
-        if (name.empty())
-        {
-            std::cerr << "Error: specific users must have an explicit name." << std::endl;
-            return false;
-        }
-
-        auto id = posix::PosixUser::getUserID(name);
-        if (!id.has_value())
-        {
-            return false;
-        }
-
-        return addPermissionEntry(category, permission, id.value());
+        std::cerr << "Error: specific users must have an explicit name." << std::endl;
```

Comment:
Not sure how we want to deal with these kind of errors. Later on we can have something like a "NonEmptyString" and the interface itself will not allow this kind of error.

This is for future designs though, basically have non-null or non-empty type variations of some of the types. I think this one here can even be templated like `NonEmpty\<string\>`. The crucial question here is always how to keep it expressive AND keep checking overhead low.

Nothing to be changed in this PR.